src/Middleware/RequestBodyParserMiddleware.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace React\Http\Middleware;
+
+use Psr\Http\Message\ServerRequestInterface;
+
+final class RequestBodyParserMiddleware
+{
+    public function __invoke(ServerRequestInterface $request, $next)
+    {
+        $type = $request->getHeaderLine('Content-Type');
+
+        if ($type === 'application/x-www-form-urlencoded') {
```

Comment:
I'm trying to find out since ever if header field **values** are case-sensitive. I'm talking only about the **values**, as header names are [clearly defined case-insensitive](https://tools.ietf.org/html/rfc7230#section-3.2). I've only found some definitions for specific header values, eg. for [`Connect`](https://tools.ietf.org/html/rfc7230#section-6.1) or [Transfer Coding names](https://tools.ietf.org/html/rfc7230#section-4).

Not sure, but it probably won't hurt to lowercase `$type` here.

Reply:
Not sure either, but I've updated the middleware forcing it to lowercase :+1:

Reply:
For the reference: This change is correct :+1: Header field values are not case-insensitive per se, but only for certain header field names, including this one:

> The type, subtype, and parameter name tokens are case-insensitive.

https://tools.ietf.org/html/rfc7231#section-3.1.1.1 and https://tools.ietf.org/html/rfc1866#section-8.2.1